### PR TITLE
SNOW-1803120:[API Coverage] type_name, simple_string, json_value, json in all data types

### PR DESCRIPTION
**Pull request by sfc-gh-yuwang**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1803120

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [x] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://docs.google.com/document/d/162d_i4zZ2AfcGRXojj0jByt8EUq-DrSHPPnTa4QvwbA/edit#bookmark=id.e82u4nekq80k)

3. Please describe how your code solves the related issue.

   This PR is meant to implement type_name, simple_string, json_value, json in VectorType, PandasDataFrameType and PandasSerieseType.
  from_json() in PandasDataFrameType and PandasSerieseType is also implemented

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,7 +12,7 @@
     - `simple_string`: Provides a simple string representation of the data.
     - `json_value`: Returns the data as a JSON-compatible value.
     - `json`: Converts the data to a JSON string.
-  - To `ArrayType`, `MapType`, `StructField`, and `StructType`:
+  - To `ArrayType`, `MapType`, `StructField`, `PandasSeriesType`, `PandasDataFrameType` and `StructType`:
```
Comment (sfc-gh-jrose) [inline CHANGELOG.md:15]: Should vector be there as well?

**Review by sfc-gh-jrose — APPROVED** [review]

File: src/snowflake/snowpark/types.py [hunk 2]
```diff
@@ -677,11 +753,13 @@ def get_snowflake_col_datatypes(self):
...
     MapType,
     StructType,
+    PandasDataFrameType,
 ]
 _all_complex_types: Dict[str, Type[Union[ArrayType, MapType, StructType]]] = {
     v.typeName(): v for v in _complex_types
 }
 
+_FIXED_VECTOR = re.compile(r"vector\(\s*(int|float)\s*,\s*(\d+)\s*\)")
 _FIXED_DECIMAL_PATTERN = re.compile(r"decimal\(\s*(\d+)\s*,\s*(\d+)\s*\)")
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/types.py:763]:
```suggestion
_FIXED_VECTOR_PATTERN = re.compile(r"vector\(\s*(int|float)\s*,\s*(\d+)\s*\)")
_FIXED_DECIMAL_PATTERN = re.compile(r"decimal\(\s*(\d+)\s*,\s*(\d+)\s*\)")
```

File: tests/unit/test_types.py [hunk 3]
```diff
@@ -1208,6 +1208,75 @@ def test_snow_type_to_dtype_str():
...
+            "vector(float,8)",
+        ),
+        (
+            PandasDataFrameType(
+                [StringType(), IntegerType(), FloatType()], ["id", "col1", "col2"]
+            ),
+            "pandas<string,int,float>",
+            '{"fields":[{"name":"id","type":"string"},{"name":"col1","type":"integer"},{"name":"col2","type":"float"}],"type":"pandasdataframe"}',
+            "pandasdataframe",
+            {
```
Comment (sfc-gh-aalam) [inline tests/unit/test_types.py:1232]: if we could name it `pandas_dataframe`, it would be more readable.

**Comment by sfc-gh-aalam** [comment]
please also fill PR description

File: CHANGELOG.md (hunk 1 above)
Comment (sfc-gh-yuwang) [reply CHANGELOG.md:16]: No. VectorType is an atomic type and does not have a from_json() function

**Review by sfc-gh-aalam — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (sfc-gh-jrose) [reply CHANGELOG.md:16]: Ah, I see this note is for from_json. Should there be a note for vectors `json_value` function?

**Pull request merged**